File: app/models/plant.rb
# A single organism living in the ground.
class Plant < Point
  DEFAULT_ICON = "/app-resources/img/icons/generic-plant.svg"
  ATTRS = %w(meta
             name
             openfarm_slug
             plant_stage
             planted_at
             water_curve_id
             spread_curve_id
             height_curve_id
             pointer_type
             radius
             depth
             x
             y
             z)
end

File: webpack/farm_designer/plant.ts
import { PlantOptions } from "./interfaces";
import { PlantPointer } from "../interfaces";

export const DEFAULT_PLANT_RADIUS = 25;

/** Factory function for Plant types. */
export function Plant(options: PlantOptions): PlantPointer {
  const openfarm_slug = options.openfarm_slug || "not-set";
  return {
    id: options.id,
    pointer_type: "Plant",
    created_at: (options.created_at || ""),
    name: (options.name || "Untitled Plant"),
    meta: {},
    x: (options.x || 0),
    y: (options.y || 0),
    z: 0,
    radius: (options.radius || DEFAULT_PLANT_RADIUS),
    openfarm_slug,
    plant_stage: "planned"
  };
}
